src/firefly/js/charts/ui/XYPlotOptions.jsx

```diff
@@ -109,16 +109,36 @@ export function resultsSuccess(callback, flds, optionParameters) {
         stretch : flds.stretch,
         nbins : (nbinsX && nbinsY) ? {x: Number(nbinsX), y: Number(nbinsY)} : undefined,
         shading: flds.shading || undefined,
-        x : { columnOrExpr : xName, error: xErr, label : xLabel, unit : xUnit, options : xOptions},
-        y : { columnOrExpr : yName, error: yErr, label : yLabel, unit : yUnit, options : yOptions},
+        x : { columnOrExpr : xName, label : xLabel, unit : xUnit, options : xOptions},
+        y : { columnOrExpr : yName, label : yLabel, unit : yUnit, options : yOptions},
         tblId,
         zoom
     }, isUndefined);
 
-    if (xErr || yErr) {
-        if (xErr) { xyPlotParams.x.error = xErr; }
-        if (yErr) { xyPlotParams.y.error = yErr; }
-    }
+    // set error related parameters
+    ['x', 'y'].forEach((axis) => {
+        const errorsType = get(flds, [`${axis}.errorsType`]);
```

Comment:
I can not use the alternative ways you have suggested, because the fieldKey contains dot in it, ex. 'x.errorsType'. Since every fieldKey is converted into an object property, this is an example of the property, which can only be accessed using the square bracket notation. 

I did it for convenience of mapping between fields and plot parameters, which are not flat as fields. In plot parameters,  x  is an object with all x-related parameters.